CMSgov/easi-app
# [NOREF] Add default handling for zero/null types in gqlgen

## Changes and Description

- Adds methods to automatically handle marshalling and unmarshalling the zero/null types that we use on our models on the backend

This should mean we don't have to constantly write resolvers that point from a zero/null-string/bool/time/int to a regular string/bool/time/int.

## How to test this change

The most important thing to test is that the behavior of `null`ability is retained. As it stands, in `main`, it doesn't actually seem like a field represented by a `zero` string (like SystemIntake.contractName, [gql](https://github.com/CMSgov/easi-app/blob/135347ee629a04985b46bd87a2a7d68fd99dc4a6/pkg/graph/schema.graphql#L1062), [model](https://github.com/CMSgov/easi-app/blob/135347ee629a04985b46bd87a2a7d68fd99dc4a6/pkg/models/system_intake.go#L137)) actually gets returned as `nil`/`null` when it's set to `""` in the database.

Really what needs to be tested & validated on this PR is that the behavior is equivalent to what we see in `main` for fields represented by the types given to us in the null/zero packages.

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
